modules/representation_management/app/services/representation_management/accredited_entity_query.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RepresentationManagement
+  class AccreditedEntityQuery
+    MAXIMUM_RESULT_COUNT = 10
+    WORD_SIMILARITY_THRESHOLD = 0.7
+
+    def initialize(query_string)
+      @query_string = query_string
+    end
+
+    def results
+      return [] if @query_string.blank?
+
+      array_results = ActiveRecord::Base.connection.exec_query(
+        ActiveRecord::Base.send(:sanitize_sql_array, [
+                                  sql_query,
+                                  {
+                                    query_string: @query_string,
+                                    threshold: WORD_SIMILARITY_THRESHOLD,
+                                    max_results: MAXIMUM_RESULT_COUNT
+                                  }
+                                ])
+      )
+
+      transform_results_to_objects(array_results)
+    end
+
+    private
+
+    # rubocop:disable Metrics/MethodLength
+    def sql_query
+      <<-SQL.squish
+        WITH combined AS (
+          SELECT
+            id,
+            full_name AS name,
+            'AccreditedIndividual' AS model_type,
+            levenshtein(full_name, :query_string) AS distance
+          FROM
+            accredited_individuals
+          WHERE
+            word_similarity(:query_string, full_name) >= :threshold
+            AND location IS NOT NULL
+
+          UNION ALL
+
+          SELECT
+            id,
+            name AS name,
+            'AccreditedOrganization' AS model_type,
+            levenshtein(name, :query_string) AS distance
+          FROM
+            accredited_organizations
+          WHERE
+            word_similarity(:query_string, name) >= :threshold
+            AND location IS NOT NULL
+        )
+        SELECT
+          id,
+          model_type,
+          distance
+        FROM
+          combined
+        ORDER BY
+          distance ASC
+        LIMIT
+          :max_results;
+      SQL
+    end
+    # rubocop:enable Metrics/MethodLength
+
+    def transform_results_to_objects(array_results)
```

Comment:
Is this method basically looking up each rep and org to get the rest of the attributes needed by the serializer? If so, can you instead include the attributes needed in the `SELECT` statements in `#sql_query` to avoid additional lookups?